### PR TITLE
feat: add metadata impl

### PR DESCRIPTION
**Pull request by wep21**
original bag2 info
```
❯ ros2 bag info /home/daisuke/rosbag2_2022_03_10-09_05_13

Files:             rosbag2_2022_03_10-09_05_13_0.db3
Bag size:          26.6 KiB
Storage id:        sqlite3
Duration:          5.422s
Start:             Mar 10 2022 09:05:13.710 (1646870713.710)
End:               Mar 10 2022 09:05:19.132 (1646870719.132)
Messages:          46
Topic information: Topic: /chatter2 | Type: std_msgs/msg/String | Count: 6 | Serialization Format: cdr
                   Topic: /chatter | Type: std_msgs/msg/String | Count: 6 | Serialization Format: cdr
                   Topic: /parameter_events | Type: rcl_interfaces/msg/ParameterEvent | Count: 0 | Serialization Format: cdr
                   Topic: /rosout | Type: rcl_interfaces/msg/Log | Count: 34 | Serialization Format: cdr
```
mcap info
```
❯ ros2 bag info /home/daisuke/demo.mcap -s mcap

Files:             /home/daisuke/demo.mcap
Bag size:          14.7 KiB
Storage id:        mcap
Duration:          5.422s
Start:             Mar 10 2022 09:05:13.710 (1646870713.710)
End:               Mar 10 2022 09:05:19.132 (1646870719.132)
Messages:          46
Topic information: Topic: /chatter2 | Type: std_msgs/msg/String | Count: 6 | Serialization Format: mcap
                   Topic: /chatter | Type: std_msgs/msg/String | Count: 6 | Serialization Format: mcap
                   Topic: /parameter_events | Type: rcl_interfaces/msg/ParameterEvent | Count: 0 | Serialization Format: mcap
                   Topic: /rosout | Type: rcl_interfaces/msg/Log | Count: 34 | Serialization Format: mcap
```

### CODE REVIEW

**Comment by wep21** [comment]
@jhurliman I'm new to mcap, so could you review this?

File: rosbag2_storage_mcap/src/mcap_storage.cpp [hunk 1]
```diff
@@ -132,11 +149,50 @@ void MCAPStorage::open(
...
+  metadata_.relative_file_paths = {get_relative_file_path()};
+  std::unordered_map<mcap::SchemaId,
+    rosbag2_storage::TopicInformation> topic_information_schema_map;
+  std::unordered_map<mcap::ChannelId,
+    rosbag2_storage::TopicInformation> topic_information_channel_map;
+  typed_record_reader_->onSchema =
+    [&topic_information_schema_map, this](const mcap::SchemaPtr schema_ptr) {
+      rosbag2_storage::TopicInformation topic_info{};
+      topic_info.topic_metadata.type = schema_ptr->name;
+      topic_info.topic_metadata.serialization_format = get_storage_identifier();
```
Comment (jhurliman) [inline rosbag2_storage_mcap/src/mcap_storage.cpp:164]: This should be `schema_ptr->encoding`

Comment (jhurliman) [inline rosbag2_storage_mcap/src/mcap_storage.cpp:164]: This should be `schema_ptr->encoding`

**Review by jhurliman — APPROVED** [review]

**Comment by jhurliman** [comment]
Looks good! I found one issue, I can fix it in a follow-up PR.

**Pull request merged**